ps_crossselling.php

```diff
@@ -148,200 +216,198 @@ public function renderForm()
                     ],
                 ],
+                'buttons' => [
+                    'clear_cache' => [
+                        'id' => 'clear_cache',
+                        'title' => $this->trans('Clear cache', [], 'Admin.Advparameters.Feature'),
+                        'icon' => 'process-icon-eraser',
+                        'href' => $this->context->link->getAdminLink(
+                            'AdminModules',
+                            true,
+                            [],
+                            [
+                                'configure' => $this->name,
+                                'submitClearCache' => 1,
+                            ]
+                        ),
+                    ],
+                ],
                 'submit' => [
                     'title' => $this->trans('Save', [], 'Admin.Actions'),
                 ],
             ],
         ];
 
-        $lang = new Language((int) Configuration::get('PS_LANG_DEFAULT'));
-
         $helper = new HelperForm();
         $helper->show_toolbar = false;
         $helper->table = $this->table;
-        $helper->default_form_language = $lang->id;
-        $helper->allow_employee_form_lang = Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') ? Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') : 0;
+        $helper->default_form_language = (int) Configuration::get('PS_LANG_DEFAULT');
+        $helper->allow_employee_form_lang = (bool) Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG');
         $helper->identifier = $this->identifier;
         $helper->submit_action = 'submitCross';
         $helper->currentIndex = $this->context->link->getAdminLink('AdminModules', false) .
-            '&configure=' . $this->name .
-            '&tab_module=' . $this->tab .
-            '&module_name=' . $this->name;
+            '&configure=' . $this->name;
         $helper->token = Tools::getAdminTokenLite('AdminModules');
         $helper->tpl_vars = [
-            'fields_value' => $this->getConfigFieldsValues(),
+            'fields_value' => [
+                'CROSSSELLING_NBR' => Tools::getValue('CROSSSELLING_NBR', (int) Configuration::get('CROSSSELLING_NBR')),
+                'CROSSSELLING_DISPLAY_PRICE' => Tools::getValue('CROSSSELLING_DISPLAY_PRICE', (bool) Configuration::get('CROSSSELLING_DISPLAY_PRICE')),
+            ],
             'languages' => $this->context->controller->getLanguages(),
-            'id_language' => $this->context->language->id,
+            'id_language' => (int) $this->context->language->id,
         ];
 
         return $helper->generateForm([$fields_form]);
     }
 
-    public function getConfigFieldsValues()
+    /**
+     * Gets Product ids used to retrieve related products already ordered
+     *
+     * @param string $hookName
+     * @param array{cookie: Cookie, cart: Cart, altern: int} $configuration Hooks arguments may varies
```

Comment:
Looks like cookie and cart are optional and there's a missing product array.
```suggestion
     * @param array{cookie?: Cookie, cart?: Cart, product: array{id_product: int}, altern: int} $configuration Hooks arguments sent to the widget
```